sockets/1.2/modules/ROOT/pages/editorial/sockets-connector-studio.adoc

```diff
@@ -0,0 +1,271 @@
+| *Advanced* | Optionally specify the reconnection information, including a reconnection strategy.
+|===
+
+[start=7]
+. Click *Test Connection* to confirm that Mule can connect with the specific server.
+. Click *OK*.
+
+Additionally, for the *Send and receive* operation, configure the MIME Type as follows:
+
+. Click the *Send and receive* operation from your flow.
+. On the *MIME Type* tab, select a MIME type from the drop-down menu field.
+
+In Studio, the Sockets Send operation and Send and receive operation in the flow look like this:
+
+.Sockets Send and Receive Operation
+image::socket-send-operation.png[Sockets Send and Receive operation]
+
+In Studio, configuring either the Send operation or the Send and receive operation, with a TCP connection to accept local incoming connections on port `8082` looks like this:
+
+.Sockets Send or Send and Receive Operation Configuration
+image::socket-sendreceive-configuration.png[Sockets Send or Send and Receive operation Configuration]
+
+In the *Configuration XML* tab, the Send operation configuration XML looks like this:
+
+[source,xml,linenums]
+----
+<sockets:request-config name="Sockets_Send_config">
+		<sockets:tcp-requester-connection host="localhost" port="8082" />
+	</sockets:request-config>
+
+	<flow name="SocketsFlow" >
+		<sockets:send config-ref="Sockets_Send_config"/>
+	</flow>
+----
+
+In Studio, configuring the MIME Type as `application/json` for the Send and receive operation looks like this:
+
+.Sockets MIME Type Configuration
+image::socket-mimetype-configuration.png[Sockets MIME type Configuration]
+
+In the *Configuration XML* tab, the Send and receive operation configuration XML looks like this:
+
+[source,xml,linenums]
+----
+<sockets:request-config name="Sockets_Send_config">
+	<sockets:tcp-requester-connection host="localhost" port="8082" />
+</sockets:request-config>
+
+<flow name="SocketsFlow" >
+	<sockets:send-and-receive config-ref="Sockets_Send_Receive_config" outputMimeType="application/json"/>
+</flow>
+----
+
+== Configure the Supported TCP Protocols
+
+Sockets Connector supports application-level protocols implemented on top of TCP. You must implement these protocols in both client and server ends so they can work properly. +
+By default, the Sockets Connector implements the Safe Protocol; however, you can configure a different protocol for the connector operations both in Anypoint Studio and XML.
+
+Supported TCP protocols:
+
+* Direct Protocol +
+  The socket reads until no more bytes are immediately available. On slow networks, `EOFProtocol` and `LengthProtocol` might be more reliable.
+* EOF Protocol +
+  Reading is terminated by the stream being closed by the client.
+* Length Protocol +
+  This protocol is defined by sending or reading an integer (the packet length) and then the data to transfer.
+* Custom Class Loading Length Protocol +
+  A length protocol that uses a specific class loader to load objects from streams.
```

Comment:
Note that at this point, the definitions move from being complete sentences to sentence fragments.
In all writing, try to make list items parallel: all fragments or all sentences.